src/basic/procfs-util.c

```diff
@@ -201,6 +201,41 @@ int procfs_cpu_get_usage(nsec_t *ret) {
         return 0;
 }
 
+int convert_meminfo_value_to_uint64_bytes(char *word, uint64_t *ret) {
+        char *digits, *e;
+        uint64_t v;
+        size_t n;
+        int r;
+
+        assert(word);
+        assert(ret);
+
+        /* Determine length of numeric value */
+        n = strspn(word, WHITESPACE);
+        digits = word + n;
+        n = strspn(digits, DIGITS);
+        if (n == 0)
+                return -EINVAL;
+        e = digits + n;
+
+        /* Ensure the line ends in " kB" */
+        n = strspn(e, WHITESPACE);
+        if (n == 0)
+                return -EINVAL;
+        if (!streq(e + n, "kB"))
+                return -EINVAL;
+
+        *e = 0;
+        r = safe_atou64(digits, &v);
+        if (r < 0)
+                return r;
+        if (v == UINT64_MAX)
+                return -EINVAL;
+
+        *ret = v * 1024U;
```

Comment:
in theory this could overflow? maybe be extra careful here and add a test first:

```c
if (v > UINT64_MAX/1024)
        return -EOVERFLOW;
```